pkg/cloudproviders/aws/aws.go

```diff
@@ -74,28 +76,45 @@ func (p *Provider) RunMetrics(ctx context.Context, statser stats.Statser) {
 // map is returned even in case of errors because it may contain partial data.
 func (p *Provider) Instance(ctx context.Context, IP ...gostatsd.Source) (map[gostatsd.Source]*gostatsd.Instance, error) {
 	instances := make(map[gostatsd.Source]*gostatsd.Instance, len(IP))
-	values := make([]*string, len(IP))
+	values := make([]string, len(IP))
 	for i, ip := range IP {
 		instances[ip] = nil // initialize map. Used for lookups to see if info for IP was requested
-		values[i] = aws.String(string(ip))
+		values[i] = string(ip)
 	}
-	input := &ec2.DescribeInstancesInput{
-		Filters: []*ec2.Filter{
-			{
-				Name:   aws.String("private-ip-address"),
-				Values: values,
-			},
+	privateIPFilter := "private-ip-address"
+	inputFilters := []ec2Types.Filter{
+		{
+			Name:   &privateIPFilter,
+			Values: values,
 		},
 	}
 
 	atomic.AddUint64(&p.describeInstanceCount, 1)
 	atomic.AddUint64(&p.describeInstanceInstances, uint64(len(IP)))
 	instancesFound := uint64(0)
 	pages := uint64(0)
+	var err error
+	var nextToken string
 
 	p.logger.WithField("ips", IP).Debug("Looking up instances")
-	err := p.Ec2.DescribeInstancesPagesWithContext(ctx, input, func(page *ec2.DescribeInstancesOutput, lastPage bool) bool {
+	hasPagesRemaining := true
+	for hasPagesRemaining {
 		pages++
+		input := &ec2.DescribeInstancesInput{
+			Filters:   inputFilters,
+			NextToken: &nextToken,
+		}
+
+		page, rawErr := p.Ec2.DescribeInstances(ctx, input)
+		nextToken = *page.NextToken
+		if rawErr != nil {
+			atomic.AddUint64(&p.describeInstanceErrors, 1)
+			hasPagesRemaining = false
+
+			if rawErr.Error() != "InvalidInstanceID.NotFound" {
+				err = fmt.Errorf("error listing AWS instances: %v", rawErr)
+			}
+		}
 		for _, reservation := range page.Reservations {
```

Comment:
if `rawErr != nil`,  `page` will be `nil` right? We should probably break the loop above

Reply:
https://github.com/aws/aws-sdk-go-v2/blob/main/service/ec2/api_op_DescribeInstances.go#L54
^ what I mean is here, the `page` is nil, and it'll panic when calling `page.Reservations`

Reply:
> follow up question there is, should we end the pagination if the expected error from eventual consistency surfaces
I thought it intends to end because `hasPagesRemaining` is set to `false`
But I don't quite get the business logic 🤔

Reply:
Ah then we should definitely break